### PR TITLE
feat: Optional arguments for getProductPlans, getPlanUsers, getPlanPayments

### LINKED ISSUE #34: typescript `getProductPlans` `productID` argument  type should be optional
Omitting `getProductPlans` argument fetches all subscirptions

### PR DESCRIPTION
**Pull request by vespasianvs**
Closes #34 

Changed typescript file to make args optional and marked them as option for the JSDoc header too.

### CODE REVIEW

File: sdk.d.ts [hunk 1]
```diff
@@ -70,38 +70,38 @@ declare class PaddleSDK {
...
      * const users = await client.getPlanUsers(123);
      */
-    getPlanUsers(planID: number): Promise<any>;
+    getPlanUsers(planID?: number): Promise<any>;
     /**
      * Get the list of payments for a subscription plan
      *
      * @method
-     * @param {number} planID
+     * @param {number} [planID]
```
Comment (avaly) [inline sdk.d.ts:97]: This needs updating in the JSDoc in the source file as well.

**Review by avaly — CHANGES_REQUESTED** [review]

File: sdk.d.ts (hunk 1 above)
Comment (vespasianvs) [reply sdk.d.ts:97]: Docs updated as requested

File: Documentation.md [hunk 2]
```diff
@@ -71,50 +71,53 @@ const coupons = await client.getProductCoupons(123);
...
 ```js
+const users = await client.getPlanUsers();
 const users = await client.getPlanUsers(123);
 ```
 <a name="PaddleSDK+getPlanPayments"></a>
 
-### client.getPlanPayments(planID) ⇒ <code>Promise</code>
-Get the list of payments for a subscription plan
+### client.getPlanPayments([planID]) ⇒ <code>Promise</code>
+Get the list of all payments or payments for a subscription plan
```
Comment (avaly) [inline Documentation.md:109]: This file should not be manually edited. It's generated by a lint staged task when the `sdk.js` file is changed.

This change here is not reflected in `sdk.js` for `getPlanPayments`. You'll need to edit that file only and regenerate both the `.d.ts` and the docs file with the tasks defined in `package.json`.

**Review by avaly — CHANGES_REQUESTED** [review]

File: Documentation.md (hunk 2 above)
Comment (vespasianvs) [reply Documentation.md:109]: Cool - done. Don't use jsdoc for my own projects so not done that before!

File: sdk.d.ts [hunk 3]
```diff
@@ -1,247 +1,250 @@
...
-     *  "custom_message" : "some custom message"
-     * 	"prices": [
-     *		"USD:19.99",
-     *		"EUR:15.99"
-     *	 ]
-     *	});
-     */
-    generatePayLink(body: any): Promise<any>;
-}
+export = PaddleSDK;
```
Comment (avaly) [inline sdk.d.ts:1]: Something is weird with this whole file being rewritten. Try to install the latest dependencies and run `yarn declaration:build`

**Review by avaly — CHANGES_REQUESTED** [review]

File: sdk.d.ts (hunk 3 above)
Comment (vespasianvs) [reply sdk.d.ts]: D'oh, I'm on Windows and had CRLF as the line ending, changed it to LF and it's fine now

**Review by avaly — APPROVED** [review]
This looks great now! Thanks for all the edits.

**Comment by avaly** [comment]
@vespasianvs Do you mind just rebasing this branch on top of the latest commits and regenerating the docs again?

**Comment by vespasianvs** [comment]
Branch rebased and docs regenerated (after some messing about by me!)

**Review by avaly — CHANGES_REQUESTED** [review]
Please remove the `yarn.lock` changes and I'll merge the PR afterwards.

**Comment by vespasianvs** [comment]
Done

**Comment by avaly** [comment]
:shipit: in v2.5.0

**Pull request merged**